src/test/java/net/snowflake/client/core/arrow/DateConverterTest.java

```diff
@@ -162,4 +165,76 @@ public void testRandomDates() throws SFException {
       }
     }
   }
+
+  @Test
+  public void testTimezoneDates() throws SFException {
+    int testDay = 16911;
+    Map<String, String> customFieldMeta = new HashMap<>();
+    customFieldMeta.put("logicalType", "DATE");
+    // test normal date
+    FieldType fieldType =
+        new FieldType(true, Types.MinorType.DATEDAY.getType(), null, customFieldMeta);
+
+    DateDayVector vector = new DateDayVector("date", fieldType, allocator);
+
+    vector.setSafe(0, testDay);
+
+    // Test JDBC_FORMAT_DATE_WITH_TIMEZONE=TRUE with different session timezones
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    ArrowVectorConverter converter = new DateConverter(vector, 0, this, true);
+    converter.setUseSessionTimezone(true);
+    converter.setSessionTimeZone(TimeZone.getTimeZone(System.getProperty("user.timezone")));
+    Object obj = converter.toObject(0);
+    Object utcObj =
+        ArrowResultUtil.getDate(
+            testDay,
+            TimeZone.getTimeZone("UTC"),
+            TimeZone.getTimeZone(System.getProperty("user.timezone")));
+
+    String tz = System.getProperty("user.timezone");
+    switch (System.getProperty("user.timezone")) {
```

Comment:
instead of using the system property in the unit tests we can pass the value here and write several test cases with different timezones given as strings